cmd/waiter/waiter.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"time"
+)
+
+// Waiter represents the actor that will wait for timeout, using a lock-file to keep it actively
+// waiting. When "done" is issued the lock-file is removed and the waiter ends.
+type Waiter struct {
+	flagValues *settings // command-line flags
+}
+
+// ErrTimeout emitted when timeout is reached.
+var ErrTimeout = errors.New("timeout waiting for condition")
+
+// save writes the lock-file with informed PID.
+func (w *Waiter) save(pid int) error {
+	f, err := os.Create(w.flagValues.lockFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	log.Printf("Saving '%d' (PID) on '%s' lock-file", pid, w.flagValues.lockFile)
+	pidStr := strconv.Itoa(pid)
+	if _, err = f.WriteString(pidStr); err != nil {
+		return err
+	}
+	return f.Sync()
```

Comment:
Would this shorter version be equivalent?

```suggestion
	return os.WriteFile(w.flagValues.lockFile, strconv.Itoa(pid), 0600)
```